### PR TITLE
Add GetUTxO to the shelley QueryLedger instance

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs [hunk 1]
```diff
@@ -468,6 +477,8 @@ encodeShelleyQuery query = case query of
       CBOR.encodeListLen 1 <> CBOR.encodeWord8 5
     GetFilteredUTxO addrs ->
       CBOR.encodeListLen 2 <> CBOR.encodeWord8 6 <> toCBOR addrs
+    GetUTxO ->
+      CBOR.encodeListLen 1 <> CBOR.encodeWord8 7
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs:481]: The corresponding case is missing in `decodeShelleyQuery`

**Comment by Jimbo4350** [comment]
> What's the reason for not using `GetFilteredUTxO Set.empty` on the node side?

`getFilteredUTxO` will return an empty `UTxO` if an empty set is passed.

**Comment by mrBliss** [comment]
> > What's the reason for not using `GetFilteredUTxO Set.empty` on the node side?
> 
> `getFilteredUTxO` will return an empty `UTxO` if an empty set is passed.

Ah :slightly_smiling_face:

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs [hunk 2]
```diff
@@ -363,6 +366,11 @@ instance TPraosCrypto c => QueryLedger (ShelleyBlock c) where
     = Nothing
   eqQuery (GetFilteredUTxO _) _
     = Nothing
+  eqQuery GetUTxO GetUTxO
+    = Nothing
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs:370]: This needs to be `Just Refl`

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**